### ISSUE TITLE
Having trouble reproduce the evaluation result for the pretrained model

### ISSUE THREAD

**Issue by liz2020** [zhaoweicai/mscnn#105]
@zhaoweicai  When I use the run_cascademscnn.m file to get the inference for the new cascade mscnn pre-trained model provided, I got plenty of prediction results for cars even when there is only one car in the gt. Is this behavior expected? The attached file shows the inference for image 000001. Moreover, after I modify and run the writeDetForEval.m and evaluate_object, the evaluation result is close to only 0.06%. I am not sure what I am doing wrong, and I only change the lines related to the path. 

The dataDir in my code might look weird because the dataset is stored in coco format, but I am sure it is pointing to the correct image directory and label directory.

![Screen Shot 2019-09-19 at 1 14 00 AM](https://user-images.githubusercontent.com/25622540/65215274-fdc62f80-da7a-11e9-9ef5-0e2ecb6db40b.png)
![Screen Shot 2019-09-19 at 1 42 00 AM](https://user-images.githubusercontent.com/25622540/65216419-cfe2ea00-da7e-11e9-9f00-9d608ac32bde.png)
![Screen Shot 2019-09-19 at 1 37 08 AM](https://user-images.githubusercontent.com/25622540/65216174-0704cb80-da7e-11e9-9da7-6e384c4670e0.png)
![Screen Shot 2019-09-19 at 1 22 34 AM](https://user-images.githubusercontent.com/25622540/65215579-fd7a6400-da7b-11e9-910d-1988e078e54b.png)

**Comment by liz2020** [comment]
The [line 87](https://github.com/zhaoweicai/mscnn/blob/534bcac5710a579d60827f192035f7eef6d8c585/examples/kitti_result/writeDetForEval.m#L87) of writeDetForEval.m assumes that the image with id equal to 'i' has its name put in the corresponding index of the train validation split file. This works when the image directory contains only the images related to the val.txt file and the names in val.txt file is sorted.

**Comment by liz2020** [comment]
The first index in the file generated by run_cascademscnn.m is not the image name, but the index for the images exists in the JPEGImages folder. Thus, it has the same assumption as the writeDetForEval.m file that the folder only contains the images listed in the val.txt / train.txt file.
